src/annotator/test/adder-test.js

```diff
@@ -217,6 +219,90 @@ describe('Adder', () => {
     });
   });
 
+  describe('#_findZindex', () => {
+    let container;
+
+    beforeEach(() => {
+      container = document.createElement('div');
+      document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+      container.remove();
+    });
+
+    it('returns default hard coded value if `document.elementsFromPoint` is not available', () => {
+      const elementsFromPointBackup = document.elementsFromPoint;
+      document.elementsFromPoint = undefined;
+      assert.strictEqual(adderCtrl._findZindex(0, 0), 32768);
+      document.elementsFromPoint = elementsFromPointBackup;
+    });
+
+    it('returns default value of 1', () => {
+      // Even if not elements are found, it returns 1
+      assert.strictEqual(adderCtrl._findZindex(-100000, -100000), 1);
+      assert.strictEqual(adderCtrl._findZindex(100000, 100000), 1);
+    });
+
+    it('returns the greatest zIndex', () => {
+      const createComponent = (left, top, zIndex, attachTo) =>
+        mount(
+          <div
+            style={{
+              position: 'absolute',
+              width: 1,
+              height: 1,
+              left,
+              top,
+              zIndex,
+            }}
+          />,
+          { attachTo }
+        );
```

Comment:
Using Enzyme here is OK, although for a simple static piece of DOM like this it would be just as easy to use Preact's `render` function directly.

```js
import { render } from 'preact';
...
render(<div ... />, container);
const rendered = container.firstChild;
```

Reply:
Good to know.